utils/interface-validator.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 Canonical
+# See LICENSE file for licensing details.
+
+"""Schema validator for `interface.yaml` files."""
+import yaml
+import glob
+import re 
+
+from enum import Enum
+from typing import List, Optional
+from pydantic import AnyHttpUrl, BaseModel, Field, ValidationError, ConfigDict
+
+class text:
+   BOLD = '\033[1m'
+   CYAN = '\033[96m'
+
+   END = '\033[0m'
+
+
+class StatusEnum(str, Enum):
+    DRAFT = "draft"
+    PUBLISHED = "published"
+    RETIRED = "retired"
+
```

Comment:
two blank lines between toplevel definitions. We should add a `fmt` tox env and a CI check to match